agent/consul/leader.go

```diff
@@ -875,6 +883,45 @@ func (s *Server) getOrCreateAutopilotConfig() *autopilot.Config {
 	return config
 }
 
+func (s *Server) bootstrapConfigEntries() error {
+	if s.config.PrimaryDatacenter != "" && s.config.PrimaryDatacenter != s.config.Datacenter {
+		// only bootstrap in the primary datacenter
+		return nil
+	}
+
+	if len(s.config.ConfigEntryBootstrap) < 1 {
+		// nothing to initialize
+		return nil
+	}
+
+	// TODO (mkeeler)- maybe figure out a way to retry this later. Right now it will require leadership change to cause the initialization to happen
```

Comment:
I think this is ok, since we only apply the config entries when they don't already exist. If we wanted to only apply this once, maybe we could add a batch operation and a nonce later but this seems fine for now.

Reply:
I think the issue is that during an upgrade with these present the first leader elected at the new version will hit this case and so _not_ bootstrap them but then nothing will cause them to be written even once the upgrade completes until the next leader election.

I think that's OK - we could note it in the upgrade details.

@mkeeler how hard would it be to make this be re-called by the leader on `consul reload`? I think we'd want that anyway because you might add them later once your servers are up and it would be gross to have to wait for (or force) a leadership change before it happens. If that were the case then it seems fine to make operators aware they would need to reload config on leader after upgrade is complete manually. (and even if they don't read the guide that seems like a thing they might try anyway).

Reply:
I made it reloadable config so now its a docs issue.

Turns out that was pretty easy. Alternatively once the leadership transfer stuff is in they could just force a leadership transfer.